### PR TITLE
Add readme to label microservice

### PR DESCRIPTION
**Pull request by abcdefgs0324**
Add a readme to describe where the code of two services is and which projects/clusters that the services are running.

/assign @jlewi 
/assign @zhenghuiwang 
/assign @hamelsmu

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/code-intelligence/51)

### CODE REVIEW

**Comment by jlewi** [comment]
@abcdefgs0324 Thanks. Can we flush this out? 

Imagine a new person with no context wants to pick up the GitHub Label project and has no prior context about how things work or where they are deployed. 

It would be great if the README fully captures everything they need to know. So lets pull in diagrams and text from some of the existing docs you wrote; such as how to do rollouts.

**Comment by abcdefgs0324** [comment]
@jlewi Sure, will do.

File: Label_Microservice/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+### Steps
+
+The original bot includes step 1 and step 2.
+If the flask app forwards traffic to the new service, the workers for the microservice will do repo-specific label predictions.
+The following describes the steps (the numbers) in the diagram.
+
+1. When an issue is filed, the GitHub payload will be sent to the flask app.
+
+1. If the issue is not selected, the flask app will predict one of three types of labels and add the label to the issue.
```
Comment (jlewi) [inline Label_Microservice/README.md]: What does "issue is not selected" mean here?

File: Label_Microservice/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+1. There is a pool of workers listening to Cloud Pub/Sub and they take one item in Cloud Pub/Sub every time.
+
+1. Workers load the specific models from Google Cloud Storage and do repo-specific predictions.
+
+
+## Project Settings
+
+We have two services now, the flask app for the original bot and the repo-specific label microservice.
+They can be in different GCP projects.
```
Comment (jlewi) [inline Label_Microservice/README.md]: Can you explain why they are currently in GCP projects? I think that's probably more an accident of history (i.e. @hamelsmu  had access to one project and you had access to another project)

File: Label_Microservice/README.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+We also create a staging cluster to run the whole services for testing.
+The following describes the GCP projects and clusters where the two services are running respectively.
+
+### Production
```
Comment (jlewi) [inline Label_Microservice/README.md]: Can you provide links to the YAML files for creating the Kubernetes resources for testing, staging, and production?

File: Label_Microservice/README.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# Repo-Specific Label Microservice
```
Comment (jlewi) [inline Label_Microservice/README.md]: Thanks @abcdefgs0324 . Could you please add the following information

1. The document you wrote for onboarding new repositories; can you port that over to a markdown doc and link to it from this doc?
1. Can you provide instructions for how you update the staging/production deployments?
    * e.g. how do you build the docker images?
    * which Kubernetes YAML files need to be updated?
    * How you apply those changes?

1. Can you provide instructions for how you control how much traffic is directed to the new model?

**Comment by abcdefgs0324** [comment]
@jlewi I revised readme according to the comments. PTAL. Thanks!

**Comment by jlewi** [comment]
Looks great thanks
/lgtm
/approve

**Pull request merged**